### PR TITLE
Fixes to XGBoost example

### PR DESCRIPTION
**Pull request by richardsliu**
- Makes test_size, shuffle, and early_stopping_rounds configurable.
- Export model file only if specified.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/560)

### CODE REVIEW

File: xgboost_ames_housing/housing.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 from sklearn.preprocessing import Imputer
 from xgboost import XGBRegressor
 
-def read_input(file_name, test_size=0.25):
+def read_input(file_name, test_size=0.25, shuffle=True):
```
Comment (zhenghuiwang) [inline xgboost_ames_housing/housing.py]: Can we make shuffle=False as the default value?

It makes the model output deterministic and some integration test may rely on it.

File: xgboost_ames_housing/housing.py [hunk 2]
```diff
@@ -103,14 +108,20 @@ def main(args):
...
           '--test-size',
           help='Fraction of training data to be reserved for test',
+          type=float,
           default=0.25
           )
+  parser.add_argument(
+          '--shuffle',
+          help='Whether to shuffle input before splitting',
+          type=bool,
+          default=True
```
Comment (zhenghuiwang) [inline xgboost_ames_housing/housing.py]: can we make the default value equals false?

**Comment by jlewi** [comment]
@richardsliu Any update on this PR?

**Comment by richardsliu** [comment]
I think this is no longer needed.
